lib/pg/instrumentation.rb

```diff
@@ -0,0 +1,158 @@
+require 'pg/instrumentation/version'
+require 'opentracing'
+
+module PG
+  module Instrumentation
+    class << self
+
+      attr_accessor :tracer
+      attr_accessor :tags
+
+      ::PG::Instrumentation.tags = {
+        'span.kind' => 'client',
+        'component' => 'pg',
+        'db.type' => 'pg'
+      }
+
+      def instrument(tracer: OpenTracing.global_tracer)
+        begin
+          require 'pg'
+        rescue LoadError
+          return
+        end
+
+        @tracer = tracer
+
+        patch_methods unless @instrumented
+        @instrumented = true
+      end
+
+      def patch_methods
+        ::PG::Connection.class_eval do
+
+          alias_method :initialize_original, :initialize
+          alias_method :exec_original, :exec
+          alias_method :exec_params_original, :exec_params
+          alias_method :prepare_original, :prepare
+          alias_method :exec_prepared_original, :exec_prepared
+
+          def initialize(*args)
+            return initialize_original(*args) if args.empty?
+
+            hash_arg = args.first.is_a?( Hash ) ? args.first : {}
+            db_name = hash_arg.fetch(:dbname, nil)
+            db_user = hash_arg.fetch(:user, nil)
+            host = hash_arg.fetch(:host, nil)
+            port = hash_arg.fetch(:port, nil)
+
+            ::PG::Instrumentation.tags['db.instance'] = db_name if db_name
```

Comment:
Are you intentionally modifying this hash instead of using copy?

Reply:
Yes, I am.

Reply:
Couldn't clients that communicate with multiple db instances report incorrect information?

Reply:
That's a potentially problem, when information is shared between functions. I am not quite sure copying would solve this problem...